axelor-supplychain/src/main/java/com/axelor/apps/supplychain/web/StockMoveController.java

```diff
@@ -124,4 +126,22 @@ public void setDefaultInvoicedPartner(ActionRequest request, ActionResponse resp
       TraceBackService.trace(response, e);
     }
   }
+
+  public void checkInvoiceStatus(ActionRequest request, ActionResponse response) {
+    try {
+      StockMove stockMove = request.getContext().asType(StockMove.class);
+      for (Invoice invoice : stockMove.getInvoiceSet()) {
+        if (invoice.getStatusSelect() == InvoiceRepository.STATUS_DRAFT
+            || invoice.getStatusSelect() == InvoiceRepository.STATUS_VALIDATED) {
+          response.setAlert(
+              String.format(
+                  I18n.get(
+                      SupplychainExceptionMessage.STOCK_MOVE_CANCEL_WRONG_INVOICE_STATUS_ALERT),
+                  stockMove.getStockMoveSeq()));
+        }
+      }
```

Comment:
Please check that invoiceSet is not empty nor null.
Also, move this code into a service in a method throwing an exception.